### DISCUSSION TITLE
Implement caret style cursors

### DISCUSSION THREAD

**Discussion by NNBnh** [helix-editor/helix#10598] [Ideas]
Split out of #1179 here is the summary:

> I have always looking for a modal terminal editor that use caret style cursors,
> Helix currently has implement Gap indexing #376 which mean the caret cursor dream is not far from becoming the reality.
>
> ...
>
> I thinks the movement will stay pretty much the same except for moving to the end of an object:
> 
> Because the implement Gap indexing, moving to the end of an object (e.g: a word) will place the cursor to the right of the last character of the object.

Here are some more explanation from [Pepper editor](https://github.com/vamolessa/pepper):

> Caret style cursors (like most text editors, cursors can move past last line character and text is always inserted to its left)
>
> ...
>
> Pepper is heavily inspired by Kakoune's selection based workflow and multiple cursors.
> However its cursors behave like caret ranges instead of block selections.
> That is, the cursor is not a one-char selection but only a visual cue to indicate the caret location.

Please consider implementing this feature, this will definitely make the keyboard experience way smother.

**Comment by pickfire** [comment]
I personally think it should be considered as opt-out via config, I tried vis which have this feature and I find it quite uncomformable.
